### DISCUSSION TITLE
mogrify.exe : mogrify.exe: UnableToOpenBlob '*.jpg': Invalid argument @ error/blob.c/OpenBlob/3536.

### DISCUSSION THREAD

**Discussion by dace93** [ImageMagick/ImageMagick#4329] [Help]
### ImageMagick version

7.1.0.9

### Operating system

Windows

### Operating system, version and so on

Windows 10 21H1

### Description

Hi!

We're using a powershell script to modify all .jpg's in a folder.
This works great if the folder does not contain apostrophes ('). 
If a folder contains a "'", mogrify does not work and exists with:
_Error: mogrify.exe : mogrify.exe: UnableToOpenBlob '*.jpg': Invalid argument @ error/blob.c/OpenBlob/3536._

For example:
Folder "C:\jane mom" works.
Folder "C:\jane's mom" does not work.

### Steps to Reproduce

You can use the following Powershell script to test:

```powershell
$file = "C:\jane's mom\image.jpg"
Write-Host $file
$location = Split-Path -Path $file
Write-Host $location
& "C:\Apps\PhotoViewerFix\mogrify.exe" -format jpg -define preserve-timestamp=true -path $location -strip *.jpg
```

### Images

use any .jpg you want to test

**Comment by dlemstra** [comment]
You also need to add quotes in the path here: `-path $location`.

**Reply by dace93** [reply]
Unfortunately, this does not work.
See below:

```powershell
PS C:\Apps\PhotoViewerFix> & "C:\Apps\PhotoViewerFix\mogrify.exe" -format jpg -define preserve-timestamp=true -path "C:\Apps\PhotoViewerFix\mom's spaghetti" -strip *.jpg

mogrify.exe : mogrify.exe: UnableToOpenBlob '*.jpg': Invalid argument @ error/blob.c/OpenBlob/3536.
At line:1 char:1
+ & "C:\Apps\PhotoViewerFix\mogrify.exe" -format jpg -define preserve-t ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (mogrify.exe: Un.../OpenBlob/3536.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
```

**Comment by snibgo** [comment]
It seems to work fine for me, with IM v7.1.0-4:
```
f:\web\im>powershell
Windows PowerShell
Copyright (C) 2014 Microsoft Corporation. All rights reserved.

PS f:\web\im> magick mogrify -path "f:\web\im\mom's spaghetti" toes9*.png
PS f:\web\im> exit

f:\web\im>dir "mom's spaghetti"
 Volume in drive F is New Volume
 Volume Serial Number is F8C5-053E

 Directory of f:\web\im\mom's spaghetti

08/10/2021  05:19    <DIR>          .
08/10/2021  05:19    <DIR>          ..
08/10/2021  05:19           320,268 toes99.png
```
Have you checked that you have files *.jpg in your current directory, which seems to be "C:\Apps\PhotoViewerFix"?
